From db1ad8ee7c3db36a7cca6e998e86719293d37838 Mon Sep 17 00:00:00 2001
From: Logan Rathbone <poprocks@gmail.com>
Date: Mon, 8 May 2023 14:39:53 -0400
Subject: [PATCH] data: Manpage generation changes

Don't automatically generate the manpage from help2man for now. This
creates problems since a display manager is required to run `zenity
--help-all` and `zenity --version` at this time. This should be fixed in
the long term, but for now we're just going to avoid the issue
altogether.

Manual intervention will be required to update the manpage for now, but
hopefully we can rectify this in the future. It is fine for now as the
manpage should not need to be regenerated too frequently.

Also, s/GTK+/GTK in the .roff boilerplate.
---
 data/README.manpage      |   6 +
 data/generate-manpage.sh |   3 +
 data/meson.build         |  23 +-
 data/zenity.1            | 496 +++++++++++++++++++++++++++++++++++++++
 data/zenity.roff         |   2 +-
 5 files changed, 508 insertions(+), 22 deletions(-)
 create mode 100644 data/README.manpage
 create mode 100755 data/generate-manpage.sh
 create mode 100644 data/zenity.1

diff --git a/data/README.manpage b/data/README.manpage
new file mode 100644
index 00000000..5ac64a35
--- /dev/null
+++ b/data/README.manpage
@@ -0,0 +1,6 @@
+In order to generate the manpage for zenity, run the script
+`generate-manpage.sh` after installing zenity.
+
+The `help2man` utility is required. When new command-line arguments are
+added or deleted, this script should be re-run manually and
+git-committed.
diff --git a/data/generate-manpage.sh b/data/generate-manpage.sh
new file mode 100755
index 00000000..84092c77
--- /dev/null
+++ b/data/generate-manpage.sh
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+help2man --no-info --name='display GTK dialogs' --version-string=' ' --section=1 --help-option='--help-all' --include='zenity.roff' --output='zenity.1' zenity
diff --git a/data/meson.build b/data/meson.build
index 6cdfea58..04cfc15d 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -22,25 +22,6 @@ if desktop_file_validate_prg.found()
   )
 endif
 
-# manpage -- generated from help2man
+# manpage
 
-help2man = find_program('help2man', required: false)
-# help2man has to run zenity to generate the man page, which it can't do if
-# we're cross building. We also can't tell help2man what exe wrapper to use.
-if help2man.found() and not meson.is_cross_build()
-  help2man_opts = [
-    '--no-info',
-    '--section=1',
-    '--help-option=--help-all',
-    '--include=@INPUT@',
-  ]
-
-  custom_target('zenity.1',
-                output: 'zenity.1',
-                input: 'zenity.roff',
-                command: [
-                  help2man, help2man_opts, '--output=@OUTPUT@', zenity
-                ],
-                install: true,
-                install_dir: join_paths(zenity_mandir, 'man1'))
-endif
+install_man('zenity.1')
diff --git a/data/zenity.1 b/data/zenity.1
new file mode 100644
index 00000000..9ddf02ff
--- /dev/null
+++ b/data/zenity.1
@@ -0,0 +1,496 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.48.5.
+.TH ZENITY "1" "May 2023" "zenity  " "User Commands"
+.SH NAME
+zenity \- display GTK dialogs
+.SH DESCRIPTION
+
+\fBzenity\fP is a program that will display GTK+ dialogs, and return
+(either in the return code, or on standard output) the users
+input. This allows you to present information, and ask for information
+from the user, from all manner of shell scripts.
+.PP
+For example, \fBzenity \-\-question\fP will return either 0, 1 or 5,
+depending on whether the user pressed \fIOK\fP, \fICancel\fP or timeout
+has been reached. \fBzenity \-\-entry\fP will output on standard output
+what the user typed into the text entry field.
+.PP
+Comprehensive documentation is also available in the GNOME Help Browser.
+.SS "Usage:"
+.IP
+zenity [OPTION?]
+.SS "Help Options:"
+.TP
+\fB\-h\fR, \fB\-\-help\fR
+Show help options
+.TP
+\fB\-\-help\-all\fR
+Show all help options
+.TP
+\fB\-\-help\-general\fR
+Show general options
+.TP
+\fB\-\-help\-calendar\fR
+Show calendar options
+.TP
+\fB\-\-help\-entry\fR
+Show text entry options
+.TP
+\fB\-\-help\-error\fR
+Show error options
+.TP
+\fB\-\-help\-info\fR
+Show info options
+.TP
+\fB\-\-help\-file\-selection\fR
+Show file selection options
+.TP
+\fB\-\-help\-list\fR
+Show list options
+.TP
+\fB\-\-help\-notification\fR
+Show notification options
+.TP
+\fB\-\-help\-progress\fR
+Show progress options
+.TP
+\fB\-\-help\-question\fR
+Show question options
+.TP
+\fB\-\-help\-warning\fR
+Show warning options
+.TP
+\fB\-\-help\-scale\fR
+Show scale options
+.TP
+\fB\-\-help\-text\-info\fR
+Show text information options
+.TP
+\fB\-\-help\-color\-selection\fR
+Show color selection options
+.TP
+\fB\-\-help\-password\fR
+Show password dialog options
+.TP
+\fB\-\-help\-forms\fR
+Show forms dialog options
+.TP
+\fB\-\-help\-misc\fR
+Show miscellaneous options
+.PP
+General options
+.TP
+\fB\-\-title\fR=\fI\,TITLE\/\fR
+Set the dialog title
+.TP
+\fB\-\-width\fR=\fI\,WIDTH\/\fR
+Set the width
+.TP
+\fB\-\-height\fR=\fI\,HEIGHT\/\fR
+Set the height
+.TP
+\fB\-\-timeout\fR=\fI\,TIMEOUT\/\fR
+Set dialog timeout in seconds
+.TP
+\fB\-\-ok\-label\fR=\fI\,TEXT\/\fR
+Set the label of the OK button
+.TP
+\fB\-\-cancel\-label\fR=\fI\,TEXT\/\fR
+Set the label of the Cancel button
+.TP
+\fB\-\-extra\-button\fR=\fI\,TEXT\/\fR
+Add an extra button
+.TP
+\fB\-\-modal\fR
+Set the modal hint
+.TP
+\fB\-\-attach\fR=\fI\,WINDOW\/\fR
+DEPRECATED; does nothing
+.PP
+Calendar options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-day\fR=\fI\,DAY\/\fR
+Set the calendar day
+.TP
+\fB\-\-month\fR=\fI\,MONTH\/\fR
+Set the calendar month
+.TP
+\fB\-\-year\fR=\fI\,YEAR\/\fR
+Set the calendar year
+.TP
+\fB\-\-date\-format\fR=\fI\,PATTERN\/\fR
+Set the format for the returned date
+.PP
+Text entry options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-entry\-text\fR=\fI\,TEXT\/\fR
+Set the entry text
+.TP
+\fB\-\-hide\-text\fR
+Hide the entry text
+.PP
+Error options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-icon\fR=\fI\,ICON\-NAME\/\fR
+Set the icon name
+.TP
+\fB\-\-no\-wrap\fR
+Do not enable text wrapping
+.TP
+\fB\-\-no\-markup\fR
+Do not enable Pango markup
+.TP
+\fB\-\-ellipsize\fR
+Enable ellipsizing in the dialog text. This fixes the high window size with long texts
+.PP
+Info options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-icon\fR=\fI\,ICON\-NAME\/\fR
+Set the icon name
+.TP
+\fB\-\-no\-wrap\fR
+Do not enable text wrapping
+.TP
+\fB\-\-no\-markup\fR
+Do not enable Pango markup
+.TP
+\fB\-\-ellipsize\fR
+Enable ellipsizing in the dialog text. This fixes the high window size with long texts
+.PP
+File selection options
+.TP
+\fB\-\-filename\fR=\fI\,FILENAME\/\fR
+Set the filename
+.TP
+\fB\-\-multiple\fR
+Allow multiple files to be selected
+.TP
+\fB\-\-directory\fR
+Activate directory\-only selection
+.TP
+\fB\-\-save\fR
+Activate save mode
+.TP
+\fB\-\-separator\fR=\fI\,SEPARATOR\/\fR
+Set output separator character
+.TP
+\fB\-\-file\-filter\fR=\fI\,NAME\/\fR | PATTERN1 PATTERN2 ...
+Set a filename filter
+.TP
+\fB\-\-confirm\-overwrite\fR
+DEPRECATED; does nothing
+.PP
+List options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-column\fR=\fI\,COLUMN\/\fR
+Set the column header
+.TP
+\fB\-\-checklist\fR
+Use check boxes for the first column
+.TP
+\fB\-\-radiolist\fR
+Use radio buttons for the first column
+.TP
+\fB\-\-imagelist\fR
+Use an image for the first column
+.TP
+\fB\-\-separator\fR=\fI\,SEPARATOR\/\fR
+Set output separator character
+.TP
+\fB\-\-multiple\fR
+Allow multiple rows to be selected
+.TP
+\fB\-\-editable\fR
+Allow changes to text
+.TP
+\fB\-\-print\-column\fR=\fI\,NUMBER\/\fR
+Print a specific column (Default is 1. 'ALL' can be used to print all columns)
+.TP
+\fB\-\-hide\-column\fR=\fI\,NUMBER\/\fR
+Hide a specific column
+.TP
+\fB\-\-hide\-header\fR
+Hide the column headers
+.TP
+\fB\-\-mid\-search\fR
+DEPRECATED; does nothing
+.PP
+Notification options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the notification text
+.TP
+\fB\-\-icon\fR=\fI\,ICON\-NAME\/\fR
+Set the icon name
+.TP
+\fB\-\-listen\fR
+Listen for commands on stdin
+.TP
+\fB\-\-hint\fR=\fI\,TEXT\/\fR
+DEPRECATED; does nothing
+.PP
+Progress options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-percentage\fR=\fI\,PERCENTAGE\/\fR
+Set initial percentage
+.TP
+\fB\-\-pulsate\fR
+Pulsate progress bar
+.TP
+\fB\-\-auto\-close\fR
+Dismiss the dialog when 100% has been reached
+.TP
+\fB\-\-auto\-kill\fR
+Kill parent process if Cancel button is pressed
+.TP
+\fB\-\-no\-cancel\fR
+Hide Cancel button
+.TP
+\fB\-\-time\-remaining\fR
+Estimate when progress will reach 100%
+.PP
+Question options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-icon\fR=\fI\,ICON\-NAME\/\fR
+Set the icon name
+.TP
+\fB\-\-no\-wrap\fR
+Do not enable text wrapping
+.TP
+\fB\-\-no\-markup\fR
+Do not enable Pango markup
+.TP
+\fB\-\-default\-cancel\fR
+Give Cancel button focus by default
+.TP
+\fB\-\-ellipsize\fR
+Enable ellipsizing in the dialog text. This fixes the high window size with long texts
+.TP
+\fB\-\-switch\fR
+Suppress OK and Cancel buttons
+.PP
+Warning options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-icon\fR=\fI\,ICON\-NAME\/\fR
+Set the icon name
+.TP
+\fB\-\-no\-wrap\fR
+Do not enable text wrapping
+.TP
+\fB\-\-no\-markup\fR
+Do not enable Pango markup
+.TP
+\fB\-\-ellipsize\fR
+Enable ellipsizing in the dialog text. This fixes the high window size with long texts
+.PP
+Scale options
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-value\fR=\fI\,VALUE\/\fR
+Set initial value
+.TP
+\fB\-\-min\-value\fR=\fI\,VALUE\/\fR
+Set minimum value
+.TP
+\fB\-\-max\-value\fR=\fI\,VALUE\/\fR
+Set maximum value
+.TP
+\fB\-\-step\fR=\fI\,VALUE\/\fR
+Set step size
+.TP
+\fB\-\-print\-partial\fR
+Print partial values
+.TP
+\fB\-\-hide\-value\fR
+Hide value
+.PP
+Text information options
+.TP
+\fB\-\-filename\fR=\fI\,FILENAME\/\fR
+Open file
+.TP
+\fB\-\-editable\fR
+Allow changes to text
+.TP
+\fB\-\-font\fR=\fI\,TEXT\/\fR
+Set the text font
+.TP
+\fB\-\-checkbox\fR=\fI\,TEXT\/\fR
+Enable an I read and agree checkbox
+.TP
+\fB\-\-auto\-scroll\fR
+Auto scroll the text to the end. Only when text is captured from stdin
+.PP
+Color selection options
+.TP
+\fB\-\-color\fR=\fI\,VALUE\/\fR
+Set the color
+.TP
+\fB\-\-show\-palette\fR
+Show the palette
+.PP
+Password dialog options
+.TP
+\fB\-\-username\fR
+Display the username option
+.PP
+Forms dialog options
+.TP
+\fB\-\-add\-entry\fR=\fI\,Field\/\fR name
+Add a new Entry in forms dialog
+.TP
+\fB\-\-add\-password\fR=\fI\,Field\/\fR name
+Add a new Password Entry in forms dialog
+.TP
+\fB\-\-add\-calendar\fR=\fI\,Calendar\/\fR field name
+Add a new Calendar in forms dialog
+.TP
+\fB\-\-add\-list\fR=\fI\,List\/\fR field and header name
+Add a new List in forms dialog
+.TP
+\fB\-\-list\-values\fR=\fI\,List\/\fR of values separated by |
+List of values for List
+.TP
+\fB\-\-column\-values\fR=\fI\,List\/\fR of values separated by |
+List of values for columns
+.TP
+\fB\-\-add\-combo\fR=\fI\,Combo\/\fR box field name
+Add a new combo box in forms dialog
+.TP
+\fB\-\-combo\-values\fR=\fI\,List\/\fR of values separated by |
+List of values for combo box
+.TP
+\fB\-\-show\-header\fR
+Show the columns header
+.TP
+\fB\-\-text\fR=\fI\,TEXT\/\fR
+Set the dialog text
+.TP
+\fB\-\-separator\fR=\fI\,SEPARATOR\/\fR
+Set output separator character
+.TP
+\fB\-\-forms\-date\-format\fR=\fI\,PATTERN\/\fR
+Set the format for the returned date
+.PP
+Miscellaneous options
+.TP
+\fB\-\-about\fR
+About zenity
+.TP
+\fB\-\-version\fR
+Print version
+.SS "Application Options:"
+.TP
+\fB\-\-calendar\fR
+Display calendar dialog
+.TP
+\fB\-\-entry\fR
+Display text entry dialog
+.TP
+\fB\-\-error\fR
+Display error dialog
+.TP
+\fB\-\-info\fR
+Display info dialog
+.TP
+\fB\-\-file\-selection\fR
+Display file selection dialog
+.TP
+\fB\-\-list\fR
+Display list dialog
+.TP
+\fB\-\-notification\fR
+Display notification
+.TP
+\fB\-\-progress\fR
+Display progress indication dialog
+.TP
+\fB\-\-question\fR
+Display question dialog
+.TP
+\fB\-\-warning\fR
+Display warning dialog
+.TP
+\fB\-\-scale\fR
+Display scale dialog
+.TP
+\fB\-\-text\-info\fR
+Display text information dialog
+.TP
+\fB\-\-color\-selection\fR
+Display color selection dialog
+.TP
+\fB\-\-password\fR
+Display password dialog
+.TP
+\fB\-\-forms\fR
+Display forms dialog
+.SH EXAMPLES
+
+Display a file selector with the title \fISelect a file to
+remove\fP. The file selected is returned on standard output.
+.IP
+zenity  \-\-title="Select a file to remove" \-\-file-selection
+.PP
+Display a text entry dialog with the title \fISelect Host\fP and the
+text \fISelect the host you would like to flood-ping\fP. The entered
+text is returned on standard output.
+.IP
+zenity  \-\-title "Select Host" \-\-entry \-\-text "Select the host you would like to flood-ping"
+.PP
+Display a dialog, asking \fIMicrosoft Windows has been found! Would
+you like to remove it?\fP. The return code will be 0 (true in shell)
+if \fIOK\fP is selected, and 1 (false) if \fICancel\fP is selected.
+.IP
+zenity  \-\-question \-\-title "Alert"  \-\-text "Microsoft Windows has been found! Would you like to remove it?"
+.PP
+Show the search results in a list dialog with the title \fISearch Results\fP
+and the text \fIFinding all header files...\fP.
+.IP
+find . \-name '*.h' | zenity \-\-list \-\-title "Search Results" \-\-text "Finding all header files.." \-\-column "Files"
+.PP
+Show a passive notification
+.IP
+zenity \-\-notification \-\-icon=software-update-available \-\-text "System update necessary!"
+.PP
+Display a weekly shopping list in a check list dialog with \fIApples\fP and \fIOranges\fP pre selected
+.IP
+zenity \-\-list \-\-checklist \-\-column "Buy" \-\-column "Item" TRUE Apples TRUE Oranges FALSE Pears FALSE Toothpaste
+.PP
+Display a progress dialog while searching for all the postscript files in your home directory
+.P
+find $HOME \-name '*.ps' | zenity \-\-progress \-\-pulsate
+.SH AUTHOR
+
+.PP
+\fBZenity\fP was originally written by Glynn Foster <glynn.foster@sun.com>. It has since been maintained by a number of contributors since then.
+.PP
+This manual page was originally written by Ross Burton <ross@burtonini.com>. It currently contains portions which are automatically generated, and other portions written by its original author.
+.SH "SEE ALSO"
+
+\fBdialog\fP(1)
diff --git a/data/zenity.roff b/data/zenity.roff
index 36d7596c..6063a7a1 100644
--- a/data/zenity.roff
+++ b/data/zenity.roff
@@ -1,6 +1,6 @@
 [description]
 
-\fBzenity\fP is a program that will display GTK+ dialogs, and return
+\fBzenity\fP is a program that will display GTK dialogs, and return
 (either in the return code, or on standard output) the users
 input. This allows you to present information, and ask for information
 from the user, from all manner of shell scripts.
-- 
GitLab

